runtime/compiler/control/CompilationThread.cpp

```diff
@@ -660,6 +660,19 @@ bool TR::CompilationInfo::shouldDowngradeCompReq(TR_MethodToBeCompiled *entry)
       TR_J9VMBase *fe = TR_J9VMBase::get(_jitConfig, NULL);
       const J9ROMMethod * romMethod = methodDetails.getRomMethod(fe);
 
+#if defined(J9VM_OPT_CRIU_SUPPORT)
+      J9VMThread *vmThread = _jitConfig->javaVM->internalVMFunctions->currentVMThread(_jitConfig->javaVM);
+      if (vmThread->javaVM->internalVMFunctions->isDebugOnRestoreEnabled(vmThread)
+          && vmThread->javaVM->internalVMFunctions->isCheckpointAllowed(vmThread)
+          && !entry->_checkpointInProgress
+          && entry->_optimizationPlan->isOptLevelDowngraded()
```

Comment:
We don't need `entry->_optimizationPlan->isOptLevelDowngraded()` and `entry->_optimizationPlan->getOptLevel() == cold` tests here. They will never be true.